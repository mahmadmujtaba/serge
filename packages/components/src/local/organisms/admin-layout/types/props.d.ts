import StatusBarTypes from '../../../molecules/statusbar/types/props'

export default interface PropTypes extends StatusBarTypes {
  /* half section title */
  type?: 'submit' | 'reset' | 'button'
  size?: 's' | 'm'
  onTabChange?: (tab: string, key: number, e: any) => void
  onPressBack?: () => void
  disabled?: boolean
  tabs?: Array<string>
  wargameChanged?: boolean
  activeTab?: string
}
