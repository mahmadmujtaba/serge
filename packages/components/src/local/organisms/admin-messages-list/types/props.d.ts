import { Message } from '@serge/custom-types'

export default interface PropTypes {
  /**
   * The messages array
   */
  messages: Array<Message>
  /*
   *  The function to perform to mark all messages as read
   */
  markAllAsRead: any
  /**
   * Current force that is seeing the messages
   */
  force: {
    name: string
    color: string
    icon?: string
  }
}
