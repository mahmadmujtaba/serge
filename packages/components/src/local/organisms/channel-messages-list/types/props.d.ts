import ChannelMessageProps from '../../../molecules/channel-message/types/props'
import ForcesInChannelProps from '../../../molecules/forces-in-channel/types/props'

interface MessagesTypes extends ChannelMessageProps {
  infoType?: boolean
  gameTurn?: number
}

export default interface PropTypes extends ForcesInChannelProps{
  /**
   * The list of channel messages properties required
   * for ChannelMessage components
   */
  messages: Array<MessagesTypes>
  /**
   * Callback on expanding message item
   */
  onRead?: { (detail: object): void }
}
