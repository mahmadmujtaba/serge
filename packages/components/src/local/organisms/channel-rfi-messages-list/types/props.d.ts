import ForcesInChannelProps from '../../../molecules/forces-in-channel/types/props'

import { MessageCustom, Role } from '@serge/custom-types'

export default interface PropTypes extends ForcesInChannelProps {
  isUmpire: boolean
  isRFIManager: boolean
  role: Role['name']
  messages: MessageCustom[]
  onChange: (nextMessage: MessageCustom) => void
  onRead: (nextMessage: MessageCustom) => void
}
