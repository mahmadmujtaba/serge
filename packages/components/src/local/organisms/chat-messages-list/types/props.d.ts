import ChatMessageProps from '../../../molecules/chat-message/types/props'
import ForcesInChannelProps from '../../../molecules/forces-in-channel/types/props'

export default interface PropTypes extends ForcesInChannelProps {
  /**
   * The list of channel messages properties required
   * for ChatMessage components
   */
  messages: Array<ChatMessageProps>
}
