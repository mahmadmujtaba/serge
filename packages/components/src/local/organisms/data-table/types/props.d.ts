import React from 'react'

export type RowDataType = string | { component: React.ReactElement | null, label: string }
export type RowWithCollapsibleType = {
  // collapsible is a function witch one will return a ReactElement and will be rendered only on call
  // cb: is a callback function with one should fire some logick on collapsible content change, ex close collapsible component
  collapsible: (cb?: () => void) => React.ReactElement
  cells: Array<RowDataType>
}
export type RowType = Array<RowDataType | RowWithCollapsibleType>
export default interface Props {
  /**
   * List of columns
   */
  columns: Array<string | { filters: Array<string>, label: string }>
  /**
   * List of data
   */
  data: Array<RowType>
}
