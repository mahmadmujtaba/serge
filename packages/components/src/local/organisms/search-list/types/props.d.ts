import { Template } from '@serge/custom-types'

type TemplateValue = Template['value']

export interface ListItem extends TemplateValue {
  details: {
    title: string
  }
}

export default interface Props {
  /**
   * Search list text input placeholder
   */
  placeholder?: string
  /**
   * Original list of searchable datas
   */
  listData: Array<ListItem>
  /**
   * Selected list item that is active on click
   */
  selected?: string
  /**
   * Callback on list item click event
   */
  setSelected: (item: ListItem) => void
}
