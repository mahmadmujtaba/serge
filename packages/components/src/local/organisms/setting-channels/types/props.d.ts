import { ChannelTypes, ForceData, TemplateBody } from '@serge/custom-types'
import { Option } from 'src/local/molecules/editable-row'

export default interface PropTypes {
  channels: Array<ChannelTypes>
  messageTemplates: Array<TemplateBody>
  forces: Array<ForceData>
  onChange: (updates: { channels: Array<ChannelTypes>, selectedChannel: ChannelTypes }) => void
  onSave?: (channel: ChannelTypes) => void
  onSidebarClick?: (selectedChannel: ChannelTypes) => void
  onCreate?: (buttonText: string | undefined) => void
  onDelete?: () => void
  onDuplicate?: () => void
  selectedChannel?: ChannelTypes
}

export type MessageGroupProps = {
  title: string
  multiple: boolean
  options?: Option[]
  onChange: (value: string[]) => void
  onDelete?: (value: string[]) => void
  type: string
  value: string[]
}

export { ChannelTypes, ChannelCollab, Channel, ForceData, Role, Participant } from '@serge/custom-types'
