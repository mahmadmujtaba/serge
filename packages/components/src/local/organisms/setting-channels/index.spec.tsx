/* global it expect */
import React from 'react'
import renderer from 'react-test-renderer'
import SettingChannels from './index'
import { ChannelData } from './types/props'
import { WargameExportedMock, MessageTemplatesMock } from '@serge/mocks'

const onChange = (updates: { channels: Array<ChannelData> }): void => { console.log(updates.channels) }

describe('SettingChannels component:', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<SettingChannels
        channels={WargameExportedMock.data.channels.channels || []}
        forces={WargameExportedMock.data.forces.forces}
        onChange={onChange}
        messages={MessageTemplatesMock}
      />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
