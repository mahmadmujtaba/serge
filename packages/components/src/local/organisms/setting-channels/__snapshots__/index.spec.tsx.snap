// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingChannels component: renders correctly 1`] = `
<div
  className="main"
>
  <div
    className="left"
  >
    <div
      className="main"
    >
      <div
        className="searchlist"
      >
        <section
          className="input-container disable-offset fullwidth"
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <div
              className="MuiInputBase-root MuiInput-root makeStyles-root-4 makeStyles-root-6 MuiInput-underline makeStyles-underline-5 makeStyles-underline-7 MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiInput-input"
                disabled={false}
                name=""
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Search"
                required={false}
                type="text"
                value=""
              />
            </div>
          </div>
        </section>
        <div
          className="searchlist-list searchlist-list"
        >
          <span
            className="searchlist-row active"
          >
            <span
              onClick={[Function]}
            >
              Channel 16
            </span>
            <div
              className="searchlist-menu"
            />
          </span>
          <span
            className="searchlist-row "
          >
            <span
              onClick={[Function]}
            >
              Blue Net
            </span>
            <div
              className="searchlist-menu"
            />
          </span>
          <span
            className="searchlist-row "
          >
            <span
              onClick={[Function]}
            >
              Mapping
            </span>
            <div
              className="searchlist-menu"
            />
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="right"
  >
    <div>
      <div
        className="row"
      >
        <div
          className="col"
        >
          <section
            className="input-container disable-offset"
          >
            <div
              className="MuiFormControl-root MuiTextField-root"
            >
              <div
                className="MuiInputBase-root MuiInput-root makeStyles-root-4 makeStyles-root-8 MuiInput-underline makeStyles-underline-5 makeStyles-underline-9 MuiInputBase-formControl MuiInput-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiInput-input"
                  disabled={false}
                  name=""
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="text"
                  value="Channel 16"
                />
              </div>
            </div>
          </section>
        </div>
        <div
          className="actions"
        >
          <button
            className="MuiButtonBase-root MuiButton-root makeStyles-root-10 makeStyles-root-12 MuiButton-contained makeStyles-contained-11 makeStyles-contained-13 MuiButton-containedSecondary"
            data-qa-type="save"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              Save Channel
            </span>
          </button>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col section table"
        >
          <div
            className="main"
          >
            <div
              className="placeholder"
            >
              Participants and messages
            </div>
            <div
              className="content"
            >
              <div
                className="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
              >
                <table
                  aria-label="simple table"
                  className="MuiTable-root"
                  role={null}
                >
                  <thead
                    className="MuiTableHead-root"
                    role={null}
                  >
                    <tr
                      className="MuiTableRow-root MuiTableRow-head"
                      role={null}
                    >
                      <th
                        aria-sort={null}
                        className="MuiTableCell-root MuiTableCell-head"
                        scope="col"
                      >
                        Force
                      </th>
                      <th
                        aria-sort={null}
                        className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
                        scope="col"
                      >
                        Restrict access to specific roles
                      </th>
                      <th
                        aria-sort={null}
                        className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
                        scope="col"
                      >
                        Templates
                      </th>
                      <th
                        aria-sort={null}
                        className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                        scope="col"
                      >
                        Actions
                      </th>
                    </tr>
                  </thead>
                  <tbody
                    className="MuiTableBody-root"
                    role={null}
                  >
                    <tr>
                      <td
                        className="td"
                      >
                        <div
                          className="input-box"
                        >
                          White
                        </div>
                      </td>
                      <td
                        className="td"
                      >
                        <div
                          className="input-box"
                        >
                          All roles
                        </div>
                      </td>
                      <td
                        className="td"
                      >
                        <div
                          className="input-box"
                        >
                          Chat if empty
                        </div>
                      </td>
                      <td
                        align="right"
                        className="td"
                      >
                        <span
                          aria-disabled={false}
                          aria-label="upload picture"
                          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
                          onBlur={[Function]}
                          onClick={[Function]}
                          onDragLeave={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          onKeyUp={[Function]}
                          onMouseDown={[Function]}
                          onMouseLeave={[Function]}
                          onMouseUp={[Function]}
                          onTouchEnd={[Function]}
                          onTouchMove={[Function]}
                          onTouchStart={[Function]}
                          role="button"
                          tabIndex={0}
                        >
                          <span
                            className="MuiIconButton-label"
                          >
                            <svg
                              aria-hidden="true"
                              className="svg-inline--fa fa-pencil-alt fa-w-16 fa-sm "
                              data-icon="pencil-alt"
                              data-prefix="fas"
                              focusable="false"
                              role="img"
                              style={Object {}}
                              viewBox="0 0 512 512"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="M497.9 142.1l-46.1 46.1c-4.7 4.7-12.3 4.7-17 0l-111-111c-4.7-4.7-4.7-12.3 0-17l46.1-46.1c18.7-18.7 49.1-18.7 67.9 0l60.1 60.1c18.8 18.7 18.8 49.1 0 67.9zM284.2 99.8L21.6 362.4.4 483.9c-2.9 16.4 11.4 30.6 27.8 27.8l121.5-21.3 262.6-262.6c4.7-4.7 4.7-12.3 0-17l-111-111c-4.8-4.7-12.4-4.7-17.1 0zM124.1 339.9c-5.5-5.5-5.5-14.3 0-19.8l154-154c5.5-5.5 14.3-5.5 19.8 0s5.5 14.3 0 19.8l-154 154c-5.5 5.5-14.3 5.5-19.8 0zM88 424h48v36.3l-64.5 11.3-31.1-31.1L51.7 376H88v48z"
                                fill="currentColor"
                                style={Object {}}
                              />
                            </svg>
                          </span>
                        </span>
                        <span
                          aria-disabled={false}
                          aria-label="upload picture"
                          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
                          onBlur={[Function]}
                          onClick={[Function]}
                          onDragLeave={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          onKeyUp={[Function]}
                          onMouseDown={[Function]}
                          onMouseLeave={[Function]}
                          onMouseUp={[Function]}
                          onTouchEnd={[Function]}
                          onTouchMove={[Function]}
                          onTouchStart={[Function]}
                          role="button"
                          tabIndex={0}
                        >
                          <span
                            className="MuiIconButton-label"
                          >
                            <svg
                              aria-hidden="true"
                              className="svg-inline--fa fa-trash fa-w-14 fa-sm "
                              data-icon="trash"
                              data-prefix="fas"
                              focusable="false"
                              role="img"
                              style={Object {}}
                              viewBox="0 0 448 512"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
                                fill="currentColor"
                                style={Object {}}
                              />
                            </svg>
                          </span>
                        </span>
                      </td>
                    </tr>
                    <tr>
                      <td
                        className="td"
                      >
                        <div
                          className="input-box"
                        >
                          Red
                        </div>
                      </td>
                      <td
                        className="td"
                      >
                        <div
                          className="input-box"
                        >
                          All roles
                        </div>
                      </td>
                      <td
                        className="td"
                      >
                        <div
                          className="input-box"
                        >
                          Chat if empty
                        </div>
                      </td>
                      <td
                        align="right"
                        className="td"
                      >
                        <span
                          aria-disabled={false}
                          aria-label="upload picture"
                          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
                          onBlur={[Function]}
                          onClick={[Function]}
                          onDragLeave={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          onKeyUp={[Function]}
                          onMouseDown={[Function]}
                          onMouseLeave={[Function]}
                          onMouseUp={[Function]}
                          onTouchEnd={[Function]}
                          onTouchMove={[Function]}
                          onTouchStart={[Function]}
                          role="button"
                          tabIndex={0}
                        >
                          <span
                            className="MuiIconButton-label"
                          >
                            <svg
                              aria-hidden="true"
                              className="svg-inline--fa fa-pencil-alt fa-w-16 fa-sm "
                              data-icon="pencil-alt"
                              data-prefix="fas"
                              focusable="false"
                              role="img"
                              style={Object {}}
                              viewBox="0 0 512 512"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="M497.9 142.1l-46.1 46.1c-4.7 4.7-12.3 4.7-17 0l-111-111c-4.7-4.7-4.7-12.3 0-17l46.1-46.1c18.7-18.7 49.1-18.7 67.9 0l60.1 60.1c18.8 18.7 18.8 49.1 0 67.9zM284.2 99.8L21.6 362.4.4 483.9c-2.9 16.4 11.4 30.6 27.8 27.8l121.5-21.3 262.6-262.6c4.7-4.7 4.7-12.3 0-17l-111-111c-4.8-4.7-12.4-4.7-17.1 0zM124.1 339.9c-5.5-5.5-5.5-14.3 0-19.8l154-154c5.5-5.5 14.3-5.5 19.8 0s5.5 14.3 0 19.8l-154 154c-5.5 5.5-14.3 5.5-19.8 0zM88 424h48v36.3l-64.5 11.3-31.1-31.1L51.7 376H88v48z"
                                fill="currentColor"
                                style={Object {}}
                              />
                            </svg>
                          </span>
                        </span>
                        <span
                          aria-disabled={false}
                          aria-label="upload picture"
                          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
                          onBlur={[Function]}
                          onClick={[Function]}
                          onDragLeave={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          onKeyUp={[Function]}
                          onMouseDown={[Function]}
                          onMouseLeave={[Function]}
                          onMouseUp={[Function]}
                          onTouchEnd={[Function]}
                          onTouchMove={[Function]}
                          onTouchStart={[Function]}
                          role="button"
                          tabIndex={0}
                        >
                          <span
                            className="MuiIconButton-label"
                          >
                            <svg
                              aria-hidden="true"
                              className="svg-inline--fa fa-trash fa-w-14 fa-sm "
                              data-icon="trash"
                              data-prefix="fas"
                              focusable="false"
                              role="img"
                              style={Object {}}
                              viewBox="0 0 448 512"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
                                fill="currentColor"
                                style={Object {}}
                              />
                            </svg>
                          </span>
                        </span>
                      </td>
                    </tr>
                    <tr>
                      <td
                        className="td"
                      >
                        <div
                          className="input-box"
                        >
                          Blue
                        </div>
                      </td>
                      <td
                        className="td"
                      >
                        <div
                          className="input-box"
                        >
                          All roles
                        </div>
                      </td>
                      <td
                        className="td"
                      >
                        <div
                          className="input-box"
                        >
                          Chat if empty
                        </div>
                      </td>
                      <td
                        align="right"
                        className="td"
                      >
                        <span
                          aria-disabled={false}
                          aria-label="upload picture"
                          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
                          onBlur={[Function]}
                          onClick={[Function]}
                          onDragLeave={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          onKeyUp={[Function]}
                          onMouseDown={[Function]}
                          onMouseLeave={[Function]}
                          onMouseUp={[Function]}
                          onTouchEnd={[Function]}
                          onTouchMove={[Function]}
                          onTouchStart={[Function]}
                          role="button"
                          tabIndex={0}
                        >
                          <span
                            className="MuiIconButton-label"
                          >
                            <svg
                              aria-hidden="true"
                              className="svg-inline--fa fa-pencil-alt fa-w-16 fa-sm "
                              data-icon="pencil-alt"
                              data-prefix="fas"
                              focusable="false"
                              role="img"
                              style={Object {}}
                              viewBox="0 0 512 512"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="M497.9 142.1l-46.1 46.1c-4.7 4.7-12.3 4.7-17 0l-111-111c-4.7-4.7-4.7-12.3 0-17l46.1-46.1c18.7-18.7 49.1-18.7 67.9 0l60.1 60.1c18.8 18.7 18.8 49.1 0 67.9zM284.2 99.8L21.6 362.4.4 483.9c-2.9 16.4 11.4 30.6 27.8 27.8l121.5-21.3 262.6-262.6c4.7-4.7 4.7-12.3 0-17l-111-111c-4.8-4.7-12.4-4.7-17.1 0zM124.1 339.9c-5.5-5.5-5.5-14.3 0-19.8l154-154c5.5-5.5 14.3-5.5 19.8 0s5.5 14.3 0 19.8l-154 154c-5.5 5.5-14.3 5.5-19.8 0zM88 424h48v36.3l-64.5 11.3-31.1-31.1L51.7 376H88v48z"
                                fill="currentColor"
                                style={Object {}}
                              />
                            </svg>
                          </span>
                        </span>
                        <span
                          aria-disabled={false}
                          aria-label="upload picture"
                          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
                          onBlur={[Function]}
                          onClick={[Function]}
                          onDragLeave={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          onKeyUp={[Function]}
                          onMouseDown={[Function]}
                          onMouseLeave={[Function]}
                          onMouseUp={[Function]}
                          onTouchEnd={[Function]}
                          onTouchMove={[Function]}
                          onTouchStart={[Function]}
                          role="button"
                          tabIndex={0}
                        >
                          <span
                            className="MuiIconButton-label"
                          >
                            <svg
                              aria-hidden="true"
                              className="svg-inline--fa fa-trash fa-w-14 fa-sm "
                              data-icon="trash"
                              data-prefix="fas"
                              focusable="false"
                              role="img"
                              style={Object {}}
                              viewBox="0 0 448 512"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
                                fill="currentColor"
                                style={Object {}}
                              />
                            </svg>
                          </span>
                        </span>
                      </td>
                    </tr>
                  </tbody>
                  <tfoot
                    className="MuiTableFooter-root"
                    role={null}
                  >
                    <tr>
                      <td
                        className="td"
                      >
                        <div
                          className="input-box"
                        >
                          <div
                            className="MuiFormControl-root"
                          >
                            <div
                              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                              onClick={[Function]}
                            >
                              <div
                                aria-haspopup="listbox"
                                aria-labelledby="mui-component-select-forces"
                                className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                                id="mui-component-select-forces"
                                onBlur={[Function]}
                                onFocus={[Function]}
                                onKeyDown={[Function]}
                                onMouseDown={[Function]}
                                role="button"
                                tabIndex={0}
                              >
                                White
                              </div>
                              <input
                                aria-hidden={true}
                                className="MuiSelect-nativeInput"
                                name="forces"
                                onAnimationStart={[Function]}
                                onChange={[Function]}
                                required={false}
                                tabIndex={-1}
                                value="0"
                              />
                              <svg
                                aria-hidden={true}
                                className="MuiSvgIcon-root MuiSelect-icon"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M7 10l5 5 5-5z"
                                />
                              </svg>
                            </div>
                          </div>
                        </div>
                      </td>
                      <td
                        className="td"
                      >
                        <div
                          className="input-box"
                        >
                          <div
                            className="MuiFormControl-root"
                          >
                            <div
                              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                              onClick={[Function]}
                            >
                              <div
                                aria-haspopup="listbox"
                                aria-labelledby="mui-component-select-access"
                                className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                                id="mui-component-select-access"
                                onBlur={[Function]}
                                onFocus={[Function]}
                                onKeyDown={[Function]}
                                onMouseDown={[Function]}
                                role="button"
                                tabIndex={0}
                              >
                                <div>
                                  All roles
                                </div>
                              </div>
                              <input
                                aria-hidden={true}
                                className="MuiSelect-nativeInput"
                                name="access"
                                onAnimationStart={[Function]}
                                onChange={[Function]}
                                required={false}
                                tabIndex={-1}
                                value=""
                              />
                              <svg
                                aria-hidden={true}
                                className="MuiSvgIcon-root MuiSelect-icon"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M7 10l5 5 5-5z"
                                />
                              </svg>
                            </div>
                          </div>
                        </div>
                      </td>
                      <td
                        className="td"
                      >
                        <div
                          className="input-box"
                        >
                          <div
                            className="MuiFormControl-root"
                          >
                            <div
                              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                              onClick={[Function]}
                            >
                              <div
                                aria-haspopup="listbox"
                                aria-labelledby="mui-component-select-templates"
                                className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                                id="mui-component-select-templates"
                                onBlur={[Function]}
                                onFocus={[Function]}
                                onKeyDown={[Function]}
                                onMouseDown={[Function]}
                                role="button"
                                tabIndex={0}
                              >
                                <div>
                                  Chat if empty
                                </div>
                              </div>
                              <input
                                aria-hidden={true}
                                className="MuiSelect-nativeInput"
                                name="templates"
                                onAnimationStart={[Function]}
                                onChange={[Function]}
                                required={false}
                                tabIndex={-1}
                                value=""
                              />
                              <svg
                                aria-hidden={true}
                                className="MuiSvgIcon-root MuiSelect-icon"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M7 10l5 5 5-5z"
                                />
                              </svg>
                            </div>
                          </div>
                        </div>
                      </td>
                      <td
                        align="right"
                        className="td"
                      >
                        <span
                          aria-disabled={false}
                          aria-label="upload picture"
                          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
                          onBlur={[Function]}
                          onClick={[Function]}
                          onDragLeave={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          onKeyUp={[Function]}
                          onMouseDown={[Function]}
                          onMouseLeave={[Function]}
                          onMouseUp={[Function]}
                          onTouchEnd={[Function]}
                          onTouchMove={[Function]}
                          onTouchStart={[Function]}
                          role="button"
                          tabIndex={0}
                        >
                          <span
                            className="MuiIconButton-label"
                          >
                            <svg
                              aria-hidden="true"
                              className="svg-inline--fa fa-undo-alt fa-w-16 "
                              data-icon="undo-alt"
                              data-prefix="fas"
                              focusable="false"
                              role="img"
                              style={Object {}}
                              viewBox="0 0 512 512"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="M255.545 8c-66.269.119-126.438 26.233-170.86 68.685L48.971 40.971C33.851 25.851 8 36.559 8 57.941V192c0 13.255 10.745 24 24 24h134.059c21.382 0 32.09-25.851 16.971-40.971l-41.75-41.75c30.864-28.899 70.801-44.907 113.23-45.273 92.398-.798 170.283 73.977 169.484 169.442C423.236 348.009 349.816 424 256 424c-41.127 0-79.997-14.678-110.63-41.556-4.743-4.161-11.906-3.908-16.368.553L89.34 422.659c-4.872 4.872-4.631 12.815.482 17.433C133.798 479.813 192.074 504 256 504c136.966 0 247.999-111.033 248-247.998C504.001 119.193 392.354 7.755 255.545 8z"
                                fill="currentColor"
                                style={Object {}}
                              />
                            </svg>
                          </span>
                        </span>
                        <span
                          aria-disabled={false}
                          aria-label="upload picture"
                          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
                          onBlur={[Function]}
                          onClick={[Function]}
                          onDragLeave={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          onKeyUp={[Function]}
                          onMouseDown={[Function]}
                          onMouseLeave={[Function]}
                          onMouseUp={[Function]}
                          onTouchEnd={[Function]}
                          onTouchMove={[Function]}
                          onTouchStart={[Function]}
                          role="button"
                          tabIndex={0}
                        >
                          <span
                            className="MuiIconButton-label"
                          >
                            <svg
                              aria-hidden="true"
                              className="svg-inline--fa fa-check fa-w-16 fa-sm "
                              data-icon="check"
                              data-prefix="fas"
                              focusable="false"
                              role="img"
                              style={Object {}}
                              viewBox="0 0 512 512"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
                                fill="currentColor"
                                style={Object {}}
                              />
                            </svg>
                          </span>
                        </span>
                      </td>
                    </tr>
                  </tfoot>
                </table>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
